Gestures: Fix flick gesture on non-desktop builds

ProfLander opened the pull request:
When building for device, the flick gesture wasn't being compiled in- these changes include and use it regardless of platform.

Open-webOS-DCO-1.0-Signed-off-by: Josh Palmer jpalmerwatkins@me.com

Commit:
Gestures: Fix flick gesture on non-desktop builds

* Remove FlickGestureRecognizer from HostQtDesktop.cpp
* Add FlickGestureRecognizer to WindowServer.cpp
* Move recognizer file entries from desktop.pri to sysmgr.pro

Open-webOS-DCO-1.0-Signed-off-by: Josh Palmer <jpalmerwatkins@me.com>

morphis committed:
MemoryMonitor: use /proc/*/oom_score_adj rather than /proc/*/oom_adj

On newer kernels /proc/*/oom_adj is deprecated and should not be used anymore. As Open
webOS requires at least a linux 3.3 kernel this changes the use to the new
/proc/*/oom_score_adj entry but still keeps backward compatiblity for older kernels which
onles provides /proc/*/oom_adj.

More information about the change are available on the following pages:
- https://github.com/mirrors/linux-2.6/commit/a63d83f427fbce97a6cea0db2e64b0eb8435cd10#include/linux/oom.h
- https://www.redhat.com/archives/lvm-devel/2011-July/msg00097.html

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>

RS102839 commented:
Thanks for the submission.

We are evaluating to make sure there aren't any unintended consequences.

ericblade-owo commented:
This does not currently build in Desktop, looks like a missing include. If you can patch that up, I can continue checking it out. Thanks!

Src/base/hosts/HostQtDesktop.cpp: In member function ‘virtual bool GestureStrip::event(QEvent_)’:
Src/base/hosts/HostQtDesktop.cpp:126:16: error: invalid use of incomplete type ‘struct QGesture’
../../luna-desktop-binaries/staging/include/QtGui/qevent.h:72:7: error: forward declaration of ‘struct QGesture’
Src/base/hosts/HostQtDesktop.cpp:127:6: error: ‘FlickGesture’ was not declared in this scope
Src/base/hosts/HostQtDesktop.cpp:127:20: error: ‘fg’ was not declared in this scope
Src/base/hosts/HostQtDesktop.cpp:127:37: error: expected type-specifier before ‘FlickGesture’
Src/base/hosts/HostQtDesktop.cpp:127:37: error: expected ‘>’ before ‘FlickGesture’
Src/base/hosts/HostQtDesktop.cpp:127:37: error: expected ‘(’ before ‘FlickGesture’
Src/base/hosts/HostQtDesktop.cpp:127:50: error: expected primary-expression before ‘>’ token
Src/base/hosts/HostQtDesktop.cpp:127:60: error: expected ‘)’ before ‘;’ token
make: *_\* [debug-x86/.obj/HostQtDesktop.o] Error 1

ericblade-owo commented:
ShiftyAxel and I have discussed this patch, and it's not something that's relevant to everywhere, but it would be nice to see it added as a compatibility option for systems that do not have operating multi-touch input, but rather use mouse emulation for the touch screen.  Hoping he will commit it with the changes we discussed for that.

morphis committed:
HostArm: close framebuffer file descriptors on shutdown

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>

morphis committed:
HostArm: skip second framebuffer initialization correctly

On some platforms we don't have a second framebuffer so until now some statements in the
init method where never reached on these platforms.

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>

morphis committed:
HostArm: initialize number of fb1 buffers correctly to zero

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>

Commit:
Update tuna configuration files to reflect new DPI scaling

Open-WebOS-DCO-1.0-Signed-Off-By: Josh Palmer <jpalmerwatkins@me.com>

morphis committed:
Correct dependency on libserviceinstaller for target devices

There is difference in the way of compiling serviceinstaller: When compiling it with cmake
it will result in a library called libserviceinstaller and if the plain Makefile is used
it will named libserviceinstall. OE uses the cmake approach and therefore we need to fix
this.

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>

morphis committed:
sound: chose correct SoundPlayer implementation based on config option

When compiling for a target device this will lead to build errors when the device has no
support for the media api.

Open-WebOS-DCO-1.0-Signed-Off-By: Simon Busch <morphis@gravedo.de>
